### PR TITLE
Add zen2 audio support

### PR DESCRIPTION
**Pull request by Emprario**
For now the code is only theoretical and it requires to be tested on every distro, especially on the distros which don't include by default Pipewire.

### CODE REVIEW

**Comment by Emprario** [comment]
Distro (all test will be done on gnome) : 
- Pop!_OS : ?
- Fedora : ?
- Ubuntu (22.10) [include by default PipeWire]: Not available yet
- Ubuntu (22.04) : ?
- Arch : ?
- Debian : ?

File: setup-audio [hunk 1]
```diff
@@ -54,7 +54,32 @@ def str_audio():
 
 
 def zen2_audio():
-    print_error("zen2 audio not implemented yet")
+    if not Path(".audio").exists(): # First setup
```
Comment (Apacelus) [inline setup-audio:57]: How is that path supposed to exist?

Also, functions.py has path_exists(), so no need to import pathlib

**Comment by Apacelus** [comment]
@Emprario Please remove the pycache file

File: setup-audio [hunk 2]
```diff
@@ -70,6 +104,8 @@ def install_package(arch_package, deb_package, rpm_package):
 
 
 if __name__ == "__main__":
+    pactl = bash("pactl info")  # pactl must run as not root
```
Comment (Apacelus) [inline setup-audio:107]: Thats not gonna work, the script is restarted when elevating to root. You'll have to save it to a file and read it, like its done with username

File: setup-audio [hunk 3]
```diff
@@ -54,7 +54,41 @@ def str_audio():
 
 
 def zen2_audio():
-    print_error("zen2 audio not implemented yet")
+    if not ("PipeWire" in pactl):
+        # Remove pulseaudio in ubuntu <= 22.04
+        # https://ubuntuhandbook.org/index.php/2022/04/pipewire-replace-pulseaudio-ubuntu-2204/
+        if path_exists("/usr/bin/apt"):
```
Comment (Apacelus) [inline setup-audio:60]: This will affect debian too.

Better read the distro name and version from /etc/eupnea.json

File: setup-audio [hunk 4]
```diff
@@ -54,7 +54,44 @@ def str_audio():
...
+                bash("ldconfig")  # auto-create links
+                bash("apt-get remove pulseaudio-module-bluetooth -y")
+                # TODO: auto run command as root
+                print_question("Please execute this command : " + "systemctl --user --now enable wireplumber.service")
+                input("Press enter after you executed the command")
+            case "arch":
+                pass
+            case "fedora":
+                pass
+            case _:
```
Comment (Apacelus) [inline setup-audio:80]: Debian?

File: setup-audio [hunk 5]
```diff
@@ -54,7 +54,44 @@ def str_audio():
...
+                print_error(f"Something went REALLY WRONG : cannot install AMD audio => Open github issue <= ")
+                exit(1)
```
Comment (Apacelus) [inline setup-audio:82]: Debian?

**Comment by Emprario** [comment]
Pop!_OS  tested => it works (update list) :+1:

**Comment by Emprario** [comment]
Ubuntu (22.04) tested => it works (update list)

**Comment by Emprario** [comment]
Fedora (37) tested => it works (update list)

**Comment by Emprario** [comment]
ArchLinux tested => it works (update list)

**Comment by Emprario** [comment]
Debian is supposed to work but installing python 3.10 on debian is a pain due to :
 - Too many errors while upgrading to unstable apt 
 - growpart that doesn't always work and debian that doesn't recognize the free space
 - => won't be tested.

**Comment by Emprario** [comment]
Ubuntu 22.10 : tested on gnome (without first time setup) => it works

**Pull request merged**